stdlib/camlinternalQuote.ml

```diff
@@ -18,4 +26,33 @@ struct
      Local { name; stamp = !currentstamp }
   | Global name | Predef { name ; _ } ->
      failwith ("CamlinternalQuote.rename " ^ name)
-end
+
+  let exec_in_current_scope scope f = let muted_vars = ref IdentSet.empty in match f () with 
+      effect FreeVar fvs, k-> let free = IdentSet.diff fvs (IdentSet.union (!muted_vars) scope) in 
+                             Printf.printf "AST has %d free variables \n" (IdentSet.cardinal fvs);
+                             Printf.printf "scope has %d bound variables \n" (IdentSet.cardinal scope);
+                             Printf.printf "There are %d free variables \n" (IdentSet.cardinal (IdentSet.diff fvs scope));
+                             (if not (IdentSet.is_empty free) then perform (FreeVar free));
+                             continue k ()
+      |effect Mute (fvs), k -> (muted_vars := IdentSet.union (!muted_vars) (fvs)); continue k ()
```

Comment:
```suggestion
      | effect Mute fvs, k -> muted_vars := IdentSet.union !muted_vars fvs; continue k ()
```

Here and throughout, please: 
- use consistent spacing (one space) after `|`
- don't wrap assignments in unnecessary parentheses
- don't use parentheses around single parameters or arguments or dereferences